src/_modules/Fs.sol

```diff
@@ -14,125 +20,339 @@ struct FsMetadata {
     /// @return Whether the file on `path` exists or not.
-    function fileExists(string memory path) internal view returns (bool) {
-        try vulcan.hevm.fsMetadata(path) {
-            return true;
+    function fileExists(string memory path) internal view returns (BoolResult) {
+        try vulcan.hevm.readFile(path) {
+            return Ok(true);
         } catch Error(string memory) {
-            return false;
+            return Ok(false);
         } catch (bytes memory reason) {
             bytes4 selector = 0x0bc44503;
-            string memory errorMessage = string.concat(
-                "The path \"", string.concat(path, "\" is not allowed to be accessed for read operations.")
-            );
+            string memory errorMessage =
+                string.concat("The path \"", path, "\" is not allowed to be accessed for read operations.");
             bytes32 errorHash = keccak256(abi.encodeWithSelector(selector, errorMessage));
             if (keccak256(reason) == errorHash) {
-                assembly {
-                    revert(add(reason, 32), mload(reason))
-                }
+                return FsErrors.Forbidden(errorMessage).toBoolResult();
             }
-            return false;
+            return Ok(false);
         }
     }
 
     /// @dev Obtains the creation code from an artifact file located at `path`
     /// @param path The file or directory to check.
     /// @return The creation bytecode.
-    function getCode(string memory path) internal view returns (bytes memory) {
-        return vulcan.hevm.getCode(path);
+    function getCode(string memory path) internal view returns (BytesResult) {
+        try vulcan.hevm.getCode(path) returns (bytes memory code) {
+            return Ok(code);
+        } catch Error(string memory reason) {
+            return FsErrors.FailedToGetCode(reason).toBytesResult();
+        } catch (bytes memory reason) {
+            return FsErrors.FailedToGetCode(abi.decode(removeSelector(reason), (string))).toBytesResult();
+        }
     }
 
     /// @dev Obtains the deployed code from an artifact file located at `path`
     /// @param path The file or directory to check.
     /// @return The deployed bytecode.
-    function getDeployedCode(string memory path) internal view returns (bytes memory) {
-        return vulcan.hevm.getDeployedCode(path);
+    function getDeployedCode(string memory path) internal view returns (BytesResult) {
+        try vulcan.hevm.getDeployedCode(path) returns (bytes memory code) {
+            return Ok(code);
+        } catch Error(string memory reason) {
+            return FsErrors.FailedToGetCode(reason).toBytesResult();
+        } catch (bytes memory reason) {
+            return FsErrors.FailedToGetCode(abi.decode(removeSelector(reason), (string))).toBytesResult();
+        }
     }
 }
+
+library FsErrors {
+    using LibError for *;
+
+    function FailedToRead(string memory reason) internal pure returns (Error) {
```

Comment:
Should these encode the reason in the error message?

Reply:
Done!